curriculum/challenges/english/25-front-end-development/lecture-understanding-error-handling/688c8ef26f3cad09f36b95b9.md

```diff
@@ -0,0 +1,268 @@
+
+## Using VS Code Debugger
+
+If you use VS Code, you can set breakpoints in your code and run the debugger to pause execution at those points. Here's how to debug the same `divide` function:
+
+**Step 1: Set up your code** Create a file called `main.py` with the following content:
+
+```py
+def divide(a, b):
+    result = a / b
+    return result
+
+print(divide(10, 2))
+print(divide(15, 3))
+```
+
+**Step 2: Set a breakpoint**
+
+1. Click in the gutter (left margin) next to line 2 (`result = a / b`) to set a breakpoint
+
+2. A red dot will appear, indicating the breakpoint is set
+
+**Step 3: Start debugging**
+
+1. Press `F5` or go to Run > Start Debugging
+
+2. Select "Python File" when prompted
+
+3. The debugger will pause execution at your breakpoint
+
+**Step 4: Inspect variables** When paused at the breakpoint, you can:
+
+- Hover over variables to see their current values
+
+- Use the Variables panel on the left to see all local variables
+
+- Use the Debug Console at the bottom to evaluate expressions
+
+**Step 5: Step through code** Use the debug toolbar to:
+
+- **Continue (F5)**: Resume execution until the next breakpoint
+
+- **Step Over (F10)**: Execute the current line and move to the next
+
+- **Step Into (F11)**: Enter into function calls
+
+- **Step Out (Shift+F11)**: Exit the current function
+
+IDE debugging tools provide a visual interface to examine the state of your program, making it easier to identify and fix issues compared to using print statements alone.
+
+By mastering these foundational debugging techniques - using print statements, the pdb module, IDE tools, and unit tests - you can effectively identify and resolve issues in your Python code. Each technique has its place: print statements for quick checks, pdb for interactive exploration, and IDE debuggers for visual inspection.
+
+# --questions--
+
+## --text--
+
+What is the purpose of using the `print()` function in debugging?
+
+## --answers--
+
+To execute code faster
+
+### --feedback--
+
+Refer back to the section where `print()` is discussed.
+
+---
+
+To display variable values and program flow
+
+### --feedback--
+
+Helps you understand the state and flow of your program.
+
+---
+
+To compile the code
+
+### --feedback--
+
+Refer back to the section where `print()` is discussed.
+
+---
+
+To terminate the program
+
+### --feedback--
+
+Refer back to the section where `print()` is discussed.
+
+## --video-solution--
+
+2
+
+## --text--
+
+What does the `pdb` module in Python provide?
+
+## --answers--
+
+A graphical user interface
+
+### --feedback--
+
+Refer back to the section where `pdb` is discussed.
+
+---
+
+A way to compile Python code
+
+### --feedback--
+
+Refer back to the section where `pdb` is discussed.
+
+---
+
+An interactive debugging environment
+
+### --feedback--
+
+Used as an interactive debugging tool.
```

Comment:
this does not seem the correct feedback, alsio this is the correct answer so it should not have a feedback. This this sentence escape from somewhere?